src/main/java/com/couchbase/lite/store/ForestDBStore.java

```diff
@@ -483,9 +484,27 @@ public RevisionList getAllRevisions(String docID, boolean onlyCurrent) {
                     int revFlags = (int) doc.getSelectedRevFlags();
                     if (((revFlags & C4RevisionFlags.kRevLeaf) != 0) == (leaf == 1 ? true : false) &&
                             RevisionInternal.generationFromRevID(revID) < generation) {
-                        revIDs.add(revID);
-                        if (outHaveBodies != null && !doc.hasRevisionBody())
+
+                        if (outHaveBodies != null && !doc.hasRevisionBody()) {
                             outHaveBodies.set(false);
+                            if (withBodiesOnly)
+                                continue;
+                        }
+                        if (withBodiesOnly) {
+                            byte[] body = new byte[0];
+                            try {
+                                body = doc.getSelectedBody();
```

Comment:
iOS also needs to call loadRevisionBody as it doesn't get loaded by default. Java doesn't need to do?

Reply:
Thanks for taking care the details. It internally calling `loadRevisionBody()` method in native layer.
cbforest:
https://github.com/couchbaselabs/cbforest/blob/86867385ef04d703cae16f9ceacec1566d960467/Java/src/com/couchbase/cbforest/Document.java#L56-L60
https://github.com/couchbaselabs/cbforest/blob/master/Java/jni/native_document.cc#L224-L234